pandas/tests/arrays/test_datetimes.py

```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
 
         roundtrip = expected.tz_localize("US/Pacific")
         tm.assert_datetime_array_equal(roundtrip, dta)
+
+    @pytest.mark.parametrize(
+        "error",
+        ["coerce", "raise"],
+    )
+    def test_fallback_different_formats(self, error):
+        # GH#46071
+        # 2 valid dates with different formats
+        # Should parse with no errors
+        s = pd.Series(["6/30/2025", "1 27 2024"])
+        expected = pd.Series(
+            [pd.Timestamp("2025-06-30 00:00:00"), pd.Timestamp("2024-01-27 00:00:00")]
+        )
+        result = pd.to_datetime(s, errors=error, infer_datetime_format=True)
+        tm.assert_series_equal(expected, result)
+
+    @pytest.mark.parametrize(
+        "dateseries",
+        [
+            pd.Series(["1/1/2000", "7/12/1200"]),
+            pd.Series(["1/1/2000", "Invalid input"]),
```

Comment:
isn't this duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L989-L1012

?

Reply:
Ok, that's a lot of comments there, so I'm going to try to answer all these questions here in this response here.  Please tell me if it's not clear which comment I'm talking about at any given point here.

I was asked about these last 4 comments before, so I'll link my previous response here so you can see it in case you missed it.
> It might look like some of these tests are reimplementations, but there are some key differences in the cases that are being tested. Given the line number you linked in that test, I assume you are mainly referring to these 3 tests in that file: `test_datetime_invalid_scalar`, `test_datetime_outofbounds_scalar`, and `test_datetime_invalid_index`. While they do share a few similarities, there are some important key differences between these tests and my added tests.
> 
> To start with, `test_datetime_invalid_scalar` and `test_datetime_outofbounds_scalar` only deal in scalar values, unlike my tests, which deal with lists. The difference from `test_datetime_invalid_index` lies largely in the data being tested. The list used in `test_datetime_invalid_index` is entirely made up of values that will not parse, while the lists in my tests always start with a valid date. This is important because `infer_datetime_format` only becomes relevant on the value after the first valid date, since it would need a format already to infer from. Therefore this test doesn't really check the fallback code as my tests are meant to.

In terms of your comments about the code changes and the first test comment, I agree with those changes and I can do that for sure.  Also, in terms of test location, I'm fully open to changing that, since the choice I made in location was largely arbitrary.

Reply:
Thanks for explaining and for your attention to detail, that's very clear! Apologies for having missed that, I'd looked through in-line comments but had missed this one in the main thread

For test location - does `pandas/tests/arrays/test_datetimes.py::TestToDatetimeInferFormat` look good to you?

Reply:
@srotondo have you had a look at https://github.com/pandas-dev/pandas/pull/48621? 

First, because you've had a close look at this part of the code so I'd really value your input

Second, because if we did go ahead with that, then it'd make the word in this PR unnecessary

Reply:
> For test location - does `pandas/tests/arrays/test_datetimes.py::TestToDatetimeInferFormat` look good to you?

To clarify, is this a new section in `test_datetimes.py` that you are suggesting, because as far as I can tell, this doesn't actually exist yet.

Reply:
I meant to add new tests to the class which

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L989-L1012

belongs to